conformance_tests/sysman/test_sysman_init/src/test_init_sysman.cpp

```diff
@@ -24,4 +67,358 @@ TEST(SysmanInitTests,
+  static char sys_env[] = "ZES_ENABLE_SYSMAN=1";
+  putenv(sys_env);
+
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zesInit(0));
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zeInit(0));
+
+  std::vector<zes_driver_handle_t> zes_drivers = get_zes_drivers();
+  std::vector<zes_device_handle_t> zes_devices =
+      get_zes_devices(zes_drivers[0]);
+  ASSERT_FALSE(zes_devices.empty());
+
+  uint32_t count = 0;
+  EXPECT_EQ(ZE_RESULT_SUCCESS,
+            zesDeviceEnumFrequencyDomains(zes_devices[0], &count, nullptr));
+}
+
+TEST(
+    SysmanInitTests,
+    GivenSysmanInitializationDoneUsingZesInitFollowedByZeInitAlongWithSysmanFlagEnabledThenWhenSysmanApiZesDeviceEnumFrequencyDomainsIsCalledWithCoreHandleThenUninitializedErrorIsReturned) {
+  static char sys_env[] = "ZES_ENABLE_SYSMAN=1";
+  putenv(sys_env);
+
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zesInit(0));
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zeInit(0));
+
+  std::vector<ze_driver_handle_t> ze_drivers = get_ze_drivers();
+  std::vector<ze_device_handle_t> ze_devices = get_ze_devices(ze_drivers[0]);
+  ASSERT_FALSE(ze_devices.empty());
+
+  uint32_t count = 0;
+  EXPECT_EQ(
+      ZE_RESULT_ERROR_UNINITIALIZED,
+      zesDeviceEnumFrequencyDomains(
+          static_cast<zes_device_handle_t>(ze_devices[0]), &count, nullptr));
+}
+
+TEST(
+    SysmanInitTests,
+    GivenCoreInitializedFirstWithSysmanFlagEnabledWhenSysmanInitializedViaZesInitThenUninitializedErrorIsReturned) {
+  static char sys_env[] = "ZES_ENABLE_SYSMAN=1";
+  putenv(sys_env);
+
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zeInit(0));
+  ASSERT_EQ(ZE_RESULT_ERROR_UNINITIALIZED, zesInit(0));
+}
+
+TEST(
+    SysmanInitTests,
+    GivenSysmanInitializedFirstViaZesInitAndCoreInitializedNextWithSysmanFlagEnabledWhenSysmanApiZesDeviceEnumFrequencyDomainsIsCalledWithCoreToSysmanMappedHandleThenSuccessIsReturned) {
+
+  static char sys_env[] = "ZES_ENABLE_SYSMAN=1";
+  putenv(sys_env);
+
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zesInit(0));
+  ASSERT_EQ(ZE_RESULT_SUCCESS, zeInit(0));
+
+  std::vector<ze_driver_handle_t> ze_drivers = get_ze_drivers();
+  std::vector<ze_device_handle_t> ze_devices = get_ze_devices(ze_drivers[0]);
+  ASSERT_FALSE(ze_devices.empty());
+
+  ze_device_properties_t properties = {ZE_STRUCTURE_TYPE_DEVICE_PROPERTIES};
+  ASSERT_EQ(ZE_RESULT_SUCCESS,
+            zeDeviceGetProperties(ze_devices[0], &properties));
+
+  zes_uuid_t uuid = {};
+  memcpy(uuid.id, properties.uuid.id, ZE_MAX_DEVICE_UUID_SIZE);
+  std::vector<zes_driver_handle_t> zes_drivers = get_zes_drivers();
+
+  zes_device_handle_t sysman_device_handle{};
+  ze_bool_t on_subdevice = false;
+  uint32_t subdevice_id = 0;
+  EXPECT_EQ(ZE_RESULT_SUCCESS, zesDriverGetDeviceByUuidExp(
+                                   zes_drivers[0], uuid, &sysman_device_handle,
+                                   &on_subdevice, &subdevice_id));
+
+  uint32_t count = 0;
```

Comment:
check for sysman_device_handle is not nullptr

Reply:
assertion added for sysman_device_handle check